repm/api_test.go

```diff
@@ -74,23 +74,31 @@ func TestBasics(t *testing.T) {
 		{"closeTransaction", `{"transactionId": 3}`, `{}`, ""},
 
 		// scan
-		{"openTransaction", `{}`, `{"transactionId":4}`, ""},
+		{"openTransaction", `{"name": "foo", "args": []}`, `{"transactionId":4}`, ""},
 		{"put", `{"transactionId": 4, "key": "foopa", "value": "doopa"}`, `{}`, ""},
-		{"commitTransaction", `{"transactionId":4}`, `{"ref":"5e14js02ptiduku9st4fgk6ovvjfe2qe"}`, ""},
+		{"commitTransaction", `{"transactionId":4}`, `{"ref":"ir8von08b95s17t04gjk6qjas4m7om78"}`, ""},
 		{"openTransaction", `{}`, `{"transactionId":5}`, ""},
 		{"scan", `{"transactionId": 5, "prefix": "foo"}`, `[{"key":"foo","value":"bar"},{"key":"foopa","value":"doopa"}]`, ""},
 		{"scan", `{"transactionId": 5, "start": {"id": {"value": "foo"}}}`, `[{"key":"foo","value":"bar"},{"key":"foopa","value":"doopa"}]`, ""},
 		{"scan", `{"transactionId": 5, "start": {"id": {"value": "foo", "exclusive": true}}}`, `[{"key":"foopa","value":"doopa"}]`, ""},
 		{"closeTransaction", `{"transactionId":5}`, `{}`, ""},
 
+		// Open transaction for replay
+		{"openTransaction", `{"name": "foo", "args": [], "rebaseOpts": {"basis": "e99uif9c7bpavajrt666es1ki52dv239", "original": "e99uif9c7bpavajrt666es1ki52dv239"}}`, ``, "only local mutations"}, // bad basis
+		{"openTransaction", `{"name": "foo", "args": [], "rebaseOpts": {"basis": "", "original": "e99uif9c7bpavajrt666es1ki52dv239"}}`, ``, "Invaild hash"},                                         // no basis
```

Comment:
Typo : "Invaild"

Reply:
funny i noticed that too but since source is here https://github.com/rocicorp/diff-server/blob/a99baedd298425ab933e2c0cb62c0b393b371161/util/noms/json/hash.go#L27 i was going to fix it up next time i touched diffserver which should be soon